src/layouts/checkout/TheCheckoutLayout.vue

```diff
@@ -0,0 +1,346 @@
+}
+
+/**
+ * Closes the the dialog that lets the user confirm (or cancel)
+ * the cancelation of the checkout and
+ * initiates the cancellation of the checkout.
+ */
+function onUserConfirmedCancellation(): void {
+    closeDialogToConfirmCancelation()
+    cancel()
+}
+
+/**
+ * Opens the the dialog that lets the user confirm (or cancel)
+ * the cancelation of the checkout if the user has not arrived at
+ * the checkout summary. If the user has actually already arrived at
+ * the checkout summary, the function simply initiates the cancellation of the checkout.
+ */
+function onUserWantsToCancel(): void {
+    if (userHasArrivedAtCheckoutSummary.value) {
+        cancel()
+    } else {
+        openDialogToConfirmCancelation()
+    }
+}
+
+/**
+ * Navigates to the shopping cart page using the router and resets the order information.
+ */
+function cancel(): void {
+    router.push({
+        name: routeNames.shoppingCart,
+    })
+    store.resetOrderToUndefined()
+}
+
+/**
+ * Proceeds to the next step in the checkout process based on the current route and button state.
+ * If the user has already arrived at the checkout summary, the function does nothing.
+ * If the current route is checkout address and the proceed button is enabled, navigates to checkout shipment.
+ * If the current route is checkout shipment and the proceed button is enabled, navigates to checkout payment.
+ * If the current route is checkout payment and the proceed button is enabled, makes the order and navigates to the order summary.
+ */
+async function proceed(): Promise<void> {
+    if (userHasArrivedAtCheckoutSummary.value) {
+        return
+    }
+
+    var whereToPushTo: string | null = null
+    if (route.name === routeNames.checkoutAddress) {
+        if (!proceedButtonDisabled.value) {
+            whereToPushTo = routeNames.checkoutShipment
+        } else {
+            return
+        }
+    } else if (route.name === routeNames.checkoutShipment) {
+        if (!proceedButtonDisabled.value) {
+            whereToPushTo = routeNames.checkoutPayment
+        } else {
+            return
+        }
+    } else if (route.name === routeNames.checkoutPayment) {
+        if (!proceedButtonDisabled.value) {
+            openConfirmOrCancelDialog()
+        }
+
+        return
+    }
+
+    if (whereToPushTo) {
+        router.push({
+            name: whereToPushTo,
+        })
+    }
+}
+
+/**
+ * Creates an order based on the current order details and user ID, then navigates to the order summary page.
+ * Additionally "clears" or resets the order information of the app store to undefined.
+ * @returns - A promise that resolves after the order is created and the navigation to the order summary page is completed.
+ */
+async function createOrderAndNavigateToOrderSummary(): Promise<void> {
+    if (currentUserId.value == undefined) {
+        return
+    }
+    isAwaitingOrderCreation.value = true
+    try {
+        var idOfCreatedOrder = await createOrder(
```

Comment:
The order also needs to be placed via the GraphQL endpoint here, the idea was to create the order before showing the summary and placing it after the user has used the confirm button on the summary page

Reply:
Ah, okay, that explains a lot! I will change it.